src/webapi/history.rs

```diff
@@ -0,0 +1,91 @@
+    /// popstate event is fired, and the state property of the event contains a copy of the history
+    /// entry's state object.
+    ///
+    /// - title — Firefox currently ignores this parameter, although it may use it in the future.
+    /// Passing the empty string here should be safe against future changes to the method.
+    /// Alternatively, you could pass a short title for the state to which you're moving.
+    ///
+    /// - URL — The new history entry's URL is given by this parameter. Note that the browser won't
+    /// attempt to load this URL after a call to pushState(), but it might attempt to load the URL
+    /// later, for instance after the user restarts the browser. The new URL does not need to be
+    /// absolute; if it's relative, it's resolved relative to the current URL. The new URL must be
+    /// of the same origin as the current URL; otherwise, pushState() will throw an exception.
+    /// This parameter is optional; if it isn't specified, it's set to the document's current URL.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_pushState%28%29_method)
+    pub fn push_state<T: JsSerializable>(&self, state: T, title: &str, url: Option<&str>) {
+        js!{ @(no_return)
+            @{self}.pushState(@{state}, @{title}, @{url});
+        };
+    }
+
+    /// Operates exactly like history.push_state() except that replace_state() modifies the current
+    /// history entry instead of creating a new one. Note that this doesn't prevent the creation of
+    /// a new entry in the global browser history.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_replaceState%28%29_method)
+    pub fn replace_state<T: JsSerializable>(&self, state: T, title: &str, url: Option<&str>) {
+        js!{ @(no_return)
+            @{self}.replaceState(@{state}, @{title}, @{url});
+        };
+    }
+
+    /// You can use the go() method to load a specific page from session history, identified by its
+    /// relative position to the current page (with the current page being, of course, relative
+    /// index 0).
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History_API#Traveling_through_history)
+    pub fn go(&self, offset: i32) {
+        js! { @(no_return)
+            @{self}.go(@{offset})
+        };
+    }
+
+    /// Move one step backward through history.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History_API#Traveling_through_history)
+    pub fn back(&self) {
+        js! { @(no_return)
+            @{self}.back()
+        };
+    }
+
+    /// Move one step forward through history.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History_API#Traveling_through_history)
+    pub fn forward(&self) {
+        js! { @(no_return)
+            @{self}.forward()
```

Comment:
Missing a semicolon here too.

Reply:
My habit of writing JS without semicolons is showing :) Fixed!